src/ramanchada2/misc/utils/matchsets.py

```diff
@@ -0,0 +1,227 @@
+import pandas as pd
+from sklearn.cluster import KMeans
+from sklearn.metrics.pairwise import euclidean_distances
+from scipy.optimize import linear_sum_assignment
+import numpy as np
+
+def match_peaks_cluster(spe_pos_dict, ref):
+    # Min-Max normalize the reference values
+    min_value = min(ref.values())
+    max_value = max(ref.values())
+    if len(ref.keys()) > 1:
+        normalized_ref = {key: (value - min_value) / (max_value - min_value) for key, value in ref.items()}
```

Comment:
should not subtract `min_value` as this way the smallest peak is getting lost. min/max makes more sense in a spectrum, not individual peaks